[OpenVINO backend] Fix and enable numpy.rot90 support

PES2UG23CS205 opened the pull request:
This PR fixes and enables numpy.rot90 support for the OpenVINO backend.

Changes include:
- Adding a proper rot90 implementation using OpenVINO transpose and reverse ops
- Validating axes and rotation parameters to match NumPy semantics
- Enabling rot90 tests by removing them from the OpenVINO excluded test list

This PR removes unrelated logspace changes to keep the scope focused.

hertschuh commented on keras/src/backend/openvino/numpy.py:
```diff
@@ -1579,7 +1622,31 @@ def logical_or(x1, x2):
+    if axis not in (0, -1):
+        raise NotImplementedError(
+            "OpenVINO logspace supports axis=0 or axis=-1 only"
+        )
```
While technically this is correct, this is now inconsistent with how `linspace` is handled.

hertschuh commented on pytest.ini:
```diff
@@ -0,0 +1,4 @@
+[pytest]
+env =
+    KERAS_BACKEND=openvino
+
```
Remove this file.

PES2UG23CS205 commented:
Thanks for the review!

I’ve updated the PR to focus only on `numpy.rot90`:
- Updated the PR title and description
- Removed all changes related to `logspace`
- Removed the local `pytest.ini`
- Enabled the `rot90` tests by removing the exclusion

All `rot90` tests are now passing locally and in CI.

Please let me know if any further changes are needed.
Thanks!

PES2UG23CS205 commented:
Thanks for the review and for pointing this out.

Understood — this PR will focus only on rot90. I will:

Restore logspace to its original state and remove any related changes

Update the PR title and description to clearly reflect rot90 support

Fix the rot90 implementation to address the OpenVINO test failures

I’m working on these changes now and will push an updated version shortly.

**While working on rot90, I noticed that OpenVINO does not expose a direct reverse operation. My current plan is to implement rot90 using a combination of reverse_sequence and transpose, following patterns used elsewhere in the OpenVINO backend.

Does this approach align with how you’d prefer rot90 to be implemented for OpenVINO, or would you recommend an alternative?**
